### DISCUSSION TITLE
ltsp image - what am I doing wrong?

### DISCUSSION THREAD

**Discussion by x0an0n** [ltsp/ltsp#295] [Q&A]
I've installed the new LTSP via PPA on Debian 10, Ubuntu 18.04, and Ubuntu 20.04 and have working chrootless clients.

However, I can't get ltsp image to work with VM files. I've made working VMs using both Virtualbox and KVM. I'm simlinking (or copying) the VM files (VMDK or raw) to /srv/ltsp and all permissions are fine. On all three distros I get the following when running ltsp image. 

I'm stuck and don't know what I'm doing wrong. 

root@multivac:/srv/ltsp# ltsp image ltsp-mate
Running: losetup -rP /dev/loop8 /srv/ltsp/ltsp-mate.img
Running: mount -t tmpfs -o mode=0755 tmpfs /tmp/tmp.9NxSVcF82n/tmpfs
Running: mount -t vfat -o ro /dev/loop8p1 /tmp/tmp.9NxSVcF82n/tmpfs/0/looproot
Running: mount -t overlay -o upperdir=/tmp/tmp.9NxSVcF82n/tmpfs/0/up,lowerdir=/tmp/tmp.9NxSVcF82n/tmpfs/0/looproot,workdir=/tmp/tmp.9NxSVcF82n/tmpfs/0/work /tmp/tmp.9NxSVcF82n/tmpfs /tmp/tmp.9NxSVcF82n/root/
**mount: /tmp/tmp.9NxSVcF82n/root: wrong fs type, bad option, bad superblock on /tmp/tmp.9NxSVcF82n/tmpfs, missing codepage or helper program, or other error.
LTSP command failed: mount -t overlay -o** upperdir=/tmp/tmp.9NxSVcF82n/tmpfs/0/up,lowerdir=/tmp/tmp.9NxSVcF82n/tmpfs/0/looproot,workdir=/tmp/tmp.9NxSVcF82n/tmpfs/0/work /tmp/tmp.9NxSVcF82n/tmpfs /tmp/tmp.9NxSVcF82n/root/
Aborting ltsp

**Comment by alkisg** [comment]
I think you're using UEFI/GPT; try BIOS/MBR for the VMs.

**Comment by x0an0n** [comment]
And I think you're quite correct!

I tried again and used manual partitioning when installing Ubuntu and everything is working. I guess automatic partitioning uses GPT by default.

Thanks for the help!

**Comment by alkisg** [comment]
> I guess automatic partitioning uses GPT by default.

I think it depends on how you booted the VM; e.g. in VirtualBox, there's an option [x] Enable UEFI mode, which is unchecked by default; I think if it's enabled, Ubuntu formats using GPT, while if it's disabled, it formats using MBR.

A way to detect if UEFI mode is enabled is to run `ls /sys/firmware/efi` inside the live session.
